integration-tests/tests/downloadWarningPage.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { AxiosResponse } from 'axios'
+import { getIntegrationTestEnvironmentVariable } from './utils/getIntegrationTestEnvironmentVariable'
+import crypto from 'crypto'
+import { TriggerEndOfFlowSQSPayload } from './utils/types/sqsPayload'
+import { invokeSQSOperationsLambda } from './utils/aws/invokeSQSOperationsLambdaFunction'
+import { sendRequest } from './utils/request/sendRequest'
+import { pollNotifyMockForDownloadUrl } from './utils/notify/pollNotifyMockForDownloadUrl'
+
+describe('Download pages', () => {
+  const assertDownloadNotFoundResponse = (response: AxiosResponse) => {
+    expect(response.status).toEqual(404)
+    expect(response.data).toEqual('')
+  }
+
+  describe('Download warning page - successful download', () => {
+    let randomId = ''
+    let fileContents = ''
+
+    beforeEach(async () => {
+      randomId = crypto.randomUUID()
+      fileContents = crypto.randomUUID()
+
+      const payload: TriggerEndOfFlowSQSPayload = {
+        message: { athenaQueryId: randomId, fileContents: fileContents },
+        queueUrl: getIntegrationTestEnvironmentVariable(
+          'INTEGRATION_TESTS_TRIGGER_QUEUE_URL'
+        )
+      }
+      await invokeSQSOperationsLambda(payload)
+    })
+
+    it('A GET should return a success response with correct max downloads when called for the first time', async () => {
+      const downloadUrl = await pollNotifyMockForDownloadUrl(randomId)
+
+      const response = await sendRequest(downloadUrl, 'GET')
+      expect(response.status).toEqual(200)
+      const contentType = response.headers['content-type']
+      expect(contentType).toEqual('text/html')
+      expect(response.data).toContain('Download the report')
+      expect(response.data).toContain('You have 3 downloads remaining.')
+    })
+
+    it('A GET should return a 404 when no record is available for the provided hash', async () => {
+      const downloadUrl = await pollNotifyMockForDownloadUrl(randomId)
```

Comment:
There's no need to set up a download here, you can just use any string as the id, or if you want, just generate another UUID (it won't be the same as another one, that's kind of the point of them) and make sure you get a 404.

Reply:
I did that to parse out the secure download base url. Is there a way to get it programmatically? My thinking was that since the table entries now had a TTL, doing this wouldn't matter much.

Reply:
The point is you don't need a real download hash for this test. Polling Notify is unnecessary and makes the test take longer.